diff --git a/xirl/losses.py b/xirl/losses.py
index 58f5a1b..34650de 100644
--- a/xirl/losses.py
+++ b/xirl/losses.py
@@ -37,7 +37,7 @@ def one_hot(y, K, smooth_eps = 0):  # pylint: disable=invalid-name
   """
   assert 0 <= smooth_eps <= 1
   assert y.ndim == 1, "Label tensor must be rank 1."
-  y_hot = torch.eye(K)[y] * (1 - smooth_eps) + (smooth_eps / (K - 1))
+  y_hot = torch.eye(K, device=y.device)[y] * (1 - smooth_eps) + (smooth_eps / (K - 1))
   return y_hot.to(y.device)
